server/sonar-server/src/main/java/org/sonar/server/metric/DefaultMetricFinder.java

```diff
@@ -0,0 +1,126 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.metric;
+
+import com.google.common.base.Function;
+import com.google.common.base.Predicate;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.sonar.api.measures.Metric;
+import org.sonar.api.measures.MetricFinder;
+import org.sonar.core.metric.db.MetricDto;
+import org.sonar.core.persistence.DbSession;
+import org.sonar.core.persistence.MyBatis;
+import org.sonar.server.db.DbClient;
+
+import static com.google.common.collect.FluentIterable.from;
+
+public class DefaultMetricFinder implements MetricFinder {
+
+  private final DbClient dbClient;
+
+  public DefaultMetricFinder(DbClient dbClient) {
+    this.dbClient = dbClient;
+  }
+
+  @Override
+  public Metric findById(int id) {
+    DbSession session = dbClient.openSession(false);
+    try {
+      MetricDto dto = dbClient.metricDao().selectNullableById(session, id);
+      if (dto != null && dto.isEnabled()) {
+        return ToMetric.INSTANCE.apply(dto);
+      }
+      return null;
+    } finally {
+      MyBatis.closeQuietly(session);
+    }
+  }
+
+  @Override
+  public Metric findByKey(String key) {
+    DbSession session = dbClient.openSession(false);
+    try {
+      MetricDto dto = dbClient.metricDao().selectNullableByKey(session, key);
+      if (dto != null && dto.isEnabled()) {
+        return ToMetric.INSTANCE.apply(dto);
+      }
+      return null;
+    } finally {
+      MyBatis.closeQuietly(session);
+    }
+  }
+
+  @Override
+  public Collection<Metric> findAll(List<String> metricKeys) {
+    DbSession session = dbClient.openSession(false);
+    try {
+      List<MetricDto> dtos = dbClient.metricDao().selectNullableByKeys(session, metricKeys);
+      return from(dtos).filter(IsEnabled.INSTANCE).transform(ToMetric.INSTANCE).toList();
+    } finally {
+      MyBatis.closeQuietly(session);
+    }
+  }
+
+  @Override
+  public Collection<Metric> findAll() {
+    DbSession session = dbClient.openSession(false);
+    try {
+      List<MetricDto> dtos = dbClient.metricDao().selectEnabled(session);
+      return from(dtos).transform(ToMetric.INSTANCE).toList();
+    } finally {
+      MyBatis.closeQuietly(session);
+    }
+  }
+
+  private enum IsEnabled implements Predicate<MetricDto> {
+    INSTANCE;
+    @Override
+    public boolean apply(@Nonnull MetricDto dto) {
+      return dto.isEnabled();
+    }
+  }
+
+  private enum ToMetric implements Function<MetricDto, Metric> {
+    INSTANCE;
+
+    @Override
+    public Metric apply(@Nonnull MetricDto dto) {
+      Metric<Serializable> metric = new Metric<>();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Constructor 'Metric(...)' is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)